docs/source/api/notebooks/Notebook_6_API2CLI.ipynb

```diff
@@ -0,0 +1,336 @@
+    "\n",
+    "If following the argument placement convension, it is easy to collect and parse them. Below we show the main entrance method of the `demo_run_train.py` file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import argparse\n",
+    "from graphstorm.config import get_argument_parser\n",
+    "\n",
+    "......\n",
+    "\n",
+    "if __name__ == '__main__':\n",
+    "    # Leverage GraphStorm's argument parser to accept configuratioin yaml file\n",
+    "    arg_parser = get_argument_parser()\n",
+    "\n",
+    "    # parse all arguments and split GraphStorm's built-in arguments from the customized ones\n",
+    "    gs_args, unknown_args = arg_parser.parse_known_args()\n",
+    "    print(f'GS arguments: {gs_args}')\n",
+    "\n",
+    "    # create a new argument parser dedicated for customized arguments\n",
+    "    cust_parser = argparse.ArgumentParser(description=\"Customized Arguments\")\n",
+    "    # add customized arguments\n",
+    "    cust_parser.add_argument('--rgat-encoder-type', type=str, default=\"ara\")\n",
+    "    cust_args = cust_parser.parse_args(unknown_args)\n",
+    "    print(f'Customized arguments: {cust_args}')\n",
+    "\n",
+    "    # use both argument sets in our main function\n",
+    "    fit(gs_args, cust_args)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "GraphStorm's config module provides a `get_argument_parser` method, which can create a argument parser, e.g., `arg_parser`, dedicated to process GraphStorm launch CLI arguments and model training and inference configurations. Using its `parse_known_args()` method, the argument parser can extract all GraphStorm built-in configurations, and also return other customized arguments, which can be processed by another argument parse, e.g., the `cust_parser`. We can then pass these arguments to the corresponding methods. Please refer to [get_argument_parser API document](https://graphstorm.readthedocs.io/en/latest/api/generated/graphstorm.config.get_argument_parser.html#graphstorm.config.get_argument_parser) for more details about this method."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## GraphStorm `GSConfig` object explanation\n",
+    "Once obtained these arguments, we can use them to create a `GSConfig` object and then pass the object to different modules to get related configurations. The `GSConfig` object can check all arguments' format and values to ensure compliance with GraphStorm specifications. Below cells show the code of creating the `GSConfig` object and examples of how to use it. Please refer to the [GSConfig API doc](https://graphstorm.readthedocs.io/en/latest/api/generated/graphstorm.config.GSConfig.html#graphstorm.config.GSConfig) for more details of this class."
```

Comment:
```suggestion
    "Once obtained these arguments, we can use them to create a `GSConfig` object and then pass the object to different modules to get related configurations. The `GSConfig` object checks every argument's format and value to ensure compliance with GraphStorm specifications. Below cells show the code of creating the `GSConfig` object and examples of how to use it. Please refer to the [GSConfig API doc](https://graphstorm.readthedocs.io/en/latest/api/generated/graphstorm.config.GSConfig.html#graphstorm.config.GSConfig) for more details of this class."
```

Reply:
Below cells show the code of creating the `GSConfig` object and examples of how to use it **to do what?**